client/cli/src/lib.rs

```diff
@@ -221,3 +253,18 @@ fn kill_color(s: &str) -> String {
 	}
 	RE.replace_all(s, "").to_string()
 }
+
+/// Generate the `cargo:` key output
+pub fn generate_cargo_keys() {
+	use vergen::{generate_cargo_keys, ConstantsFlags};
+
+	generate_cargo_keys(ConstantsFlags::SHA_SHORT).expect("Failed to generate metadata files");
+}
+
+/// Make sure the calling `build.rs` script is rerun when `.git/HEAD` changed.
+///
+/// The file is searched from the `CARGO_MANIFEST_DIR` upwards. If the file can not be found,
+/// a warning is generated.
+pub fn rerun_if_git_head_changed() {
+	build_script_utils::rerun_if_git_head_changed();
```

Comment:
Do we still need to implement this ourselves? This functionality exist already in vergen: https://docs.rs/vergen/3.1.0/vergen/struct.ConstantsFlags.html#associatedconstant.REBUILD_ON_HEAD_CHANGE

Reply:
> yes we do. Our script does a parent-search for the .git/head, while the one in vergen might only work if the build.rs is in the folder the .git is, if I understand this correctly.